cmake/DetectArchitecture.cmake

```diff
@@ -43,7 +50,7 @@ set(AVX_TYPE "AVX0" PARENT_SCOPE)
 set(AVX_FLAGS "" PARENT_SCOPE)
 
 # This is only supported on x86/x64, it is completely skipped and forced to T_fallback anywhere else
-if ((${CMAKE_SYSTEM_PROCESSOR} MATCHES "x86_64") OR (${CMAKE_SYSTEM_PROCESSOR} MATCHES "i386") OR (${CMAKE_SYSTEM_PROCESSOR} MATCHES "AMD64"))
+if ((${CMAKE_SYSTEM_PROCESSOR} MATCHES "x86_64") OR (${CMAKE_SYSTEM_PROCESSOR} MATCHES "i386") OR (${CMAKE_SYSTEM_PROCESSOR} MATCHES "AMD64") OR (${CMAKE_HOST_SYSTEM_PROCESSOR} MATCHES "arm64"))
```

Comment:
what about arm7hf? arm7hf can have neon

Reply:
https://stackoverflow.com/a/70498851
According to this, these are the possible values for this value - how do we detect if it's arm7hf?

Reply:
```
brain@raspberrypi:~ $ uname -m
armv7l
```

Looks like `armv7l` is what you need to look for